### PR TITLE
Bugfix for `isort` version

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

`isort` has experienced a bad pin which is breaking `pre-commit`, see PyCQA/isort#2079. This was fixed in version 5.11.5.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -70,7 +70,7 @@ repos:
           - tomli
 
   - repo: https://github.com/pycqa/isort
-    rev: 5.11.4
+    rev: 5.11.5
```
Comment (eerovaher) [inline .pre-commit-config.yaml:73]: 5.11.5 was for projects that still support Python 3.7, we should be able to use 5.12.0. See also PyCQA/isort#2083
```suggestion
    rev: 5.12.0
```

**Comment by pllim** [comment]
Thanks! Can we squash? 😬 🙏

**Review by pllim — APPROVED** [review]
Thanks for the quick turnaround!

The pre-commit job is green. There is no need to wait for full CI, so I am merging.

**Comment by pllim** [comment]
This affects release branches too so have to backport. 🤞 

@meeseeksdev backport to v5.0.x

@meeseeksdev backport to v5.2.x

**Comment by pllim** [comment]
I am manually backporting this as part of https://github.com/astropy/astropy/pull/14340 to v5.2.x. There is no need for v5.0.x.

**Pull request merged**